src/els_text_synchronization.erl

```diff
@@ -11,12 +13,24 @@ did_open(Params) ->
   Uri          = maps:get(<<"uri">>         , TextDocument),
   Text         = maps:get(<<"text">>        , TextDocument),
   Document     = els_dt_document:new(Uri, Text),
-  ok           = els_indexer:index(Document).
+  ok           = els_indexer:index(Document),
+  generate_diagnostics(Uri).
 
 -spec did_save(map()) -> ok.
 did_save(Params) ->
   TextDocument = maps:get(<<"textDocument">>, Params),
   Uri          = maps:get(<<"uri">>         , TextDocument),
+  generate_diagnostics(Uri).
+
+-spec did_close(map()) -> ok.
+did_close(_Params) -> ok.
+
+%%==============================================================================
+%% Internal functions
+%%==============================================================================
+
+-spec generate_diagnostics(uri()) -> ok.
+generate_diagnostics(Uri) ->
```

Comment:
I'm not sure where those message are coming from. But the function `els_text_synchronization:did_save/1` is spawned from the `els_methods:textdocument_didsave/2` function. I think we should do the same (at least for generating the diagnostics) when we call the `generate_diagnostics/1` function during `els_text_synchronization:did_open/1`.

Reply:
I agree. This should be done asynchronously.

Reply:
Great. I thought it would make sense to do that, but wanted feedback.

Reply:
Will do.